x-pack/plugins/ml/public/application/timeseriesexplorer/components/timeseriesexplorer_controls/timeseriesexplorer_controls.tsx

```diff
@@ -0,0 +1,230 @@
+  showForecastCheckbox: boolean;
+  showForecast: boolean;
+  showModelBoundsCheckbox: boolean;
+  showModelBounds: boolean;
+  toggleShowModelBoundsHandler: () => void;
+  toggleShowAnnotationsHandler: () => void;
+  toggleShowForecastHandler: () => void;
+}
+
+const SavedObjectSaveModalDashboard = withSuspense(LazySavedObjectSaveModalDashboard);
+
+function getDefaultEmbeddablePanelConfig(jobId: JobId, queryString?: string) {
+  return {
+    title: getDefaultSingleMetricViewerPanelTitle(jobId).concat(
+      queryString ? `- ${queryString}` : ''
+    ),
+    id: htmlIdGenerator()(),
+  };
+}
+
+export const TimeSeriesExplorerControls: FC<Props> = ({
+  selectedDetectorIndex,
+  selectedEntities,
+  selectedJobId,
+  showAnnotations,
+  showAnnotationsCheckbox,
+  showForecast,
+  showForecastCheckbox,
+  showModelBounds,
+  showModelBoundsCheckbox,
+  toggleShowAnnotationsHandler,
+  toggleShowModelBoundsHandler,
+  toggleShowForecastHandler,
+}) => {
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
+  const [createInDashboard, setCreateInDashboard] = useState<boolean>(false);
+
+  const {
+    services: {
+      application: { capabilities },
+      embeddable,
+    },
+  } = useMlKibana();
+
+  const canEditDashboards = capabilities.dashboard?.createNew ?? false;
+
+  const closePopoverOnAction = useCallback(
+    (actionCallback: Function) => {
+      return () => {
+        setIsMenuOpen(false);
+        actionCallback();
+      };
+    },
+    [setIsMenuOpen]
+  );
+
+  const menuPanels: EuiContextMenuPanelDescriptor[] = [
+    {
+      id: 0,
+      items: [
+        {
+          name: (
+            <FormattedMessage
+              id="xpack.ml.timeseriesExplorer.addToDashboardLabel"
+              defaultMessage="Add to dashboard"
+            />
+          ),
+          onClick: closePopoverOnAction(() => {
+            setCreateInDashboard(true);
+          }),
+        },
+      ],
+    },
+  ];
+
+  const onSaveCallback: SaveModalDashboardProps['onSave'] = useCallback(
+    ({ dashboardId, newTitle, newDescription }) => {
+      const stateTransfer = embeddable!.getStateTransfer();
+      const config = getDefaultEmbeddablePanelConfig(selectedJobId);
+
+      const embeddableInput: Partial<SingleMetricViewerEmbeddableState> = {
+        id: config.id,
+        title: newTitle,
+        description: newDescription,
+        jobIds: [selectedJobId],
+        selectedDetectorIndex,
+        selectedEntities,
+      };
+
+      const state = {
+        input: embeddableInput,
+        type: ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE,
+      };
+
+      const path = dashboardId === 'new' ? '#/create' : `#/view/${dashboardId}`;
+
+      stateTransfer.navigateToWithEmbeddablePackage('dashboards', {
+        state,
+        path,
+      });
+    },
+    [embeddable, selectedJobId, selectedDetectorIndex, selectedEntities]
+  );
+
+  return (
+    <>
+      <EuiFlexGroup style={{ float: 'right' }} alignItems="center">
```

Comment:
I know you just moved this code, but I wonder if we can get rid of the inline styles? Do we need the float here?